src/client/session/security/authenticators/sasl/mechanisms/scram/scram-mechanism.ts

```diff
@@ -8,20 +8,17 @@ import { promisify } from 'util';
 const asyncRandomBytes = promisify(randomBytes);
 const asyncPbkdf2 = promisify(pbkdf2);
 
-export class ScramAuthentication {
+export class ScramMechanism {
   private static readonly GS2_HEADER = 'n,,';
 
   private constructor(
     private readonly credentials: ScramCredentials,
-    private readonly mechanism: ScramMechanism,
+    private readonly mechanism: HashFunction,
```

Comment:
I would suggest having consistent naming between properties and their types. Either rename the `HashFunction` to `HashMechanism` or rename the property to `hashFunction`, where I personally like the second option more.